### DISCUSSION TITLE
RSF overfitting

### DISCUSSION THREAD

**Discussion by LukasBocek98** [kogalur/randomForestSRC#420] [Q&A] [answered]
Hi everyone, I wanted to ask if anyone is experiencing quite significant overfitting with random survival forests? For my master thesis I am analysing a real dataset with about 11k unique patients out of which over 5k experienced an event, I can get concordance index on my training data around 0.83 with RSF. However, on my testing data, the concordance is about 0.65, which is even worse than Cox model... (Cox got 0.67 on both, training and testing data). I tried decreasing the nodedepth and also change values for mtry and nodesize, neither of that was really helpful...I even decided to create simulated dataset and the story there was very similar...

I will be really glad for any suggestions/your experiences etc. Thank you very much! :)

**Comment by ishwaran** [comment]
No, we never experienced this.  We have looked at real and simulated data and don't see this behavior.

How are you measuring the C-index?  Can you show the output from the trained forest?  Also show the output from the prediction on the test data.  In other words if "o" is the trained object, show the output from `print(o)` and if "p" is the prediction object, show the output `print(p)`

**Reply by LukasBocek98** [reply]
Thank you for the fast reply! To measure C index I used the `pec` package, but also check with the `get.cindex`, which yields the same results. I am posting screenshots from the trained and predicted object.

 The OOB is similar to the result on test set, but on the seen data, the model was performing way better...

   
   
<img width="339" alt="Screenshot 2024-03-28 at 21 41 33" src="https://github.com/kogalur/randomForestSRC/assets/135209247/6ed2e46c-5816-4e0e-8a76-1131c8b7e74c">
<img width="286" alt="Screenshot 2024-03-28 at 21 41 48" src="https://github.com/kogalur/randomForestSRC/assets/135209247/492babd5-c550-4a07-a20c-bd784067d060">

**Comment by ishwaran** [comment]
The C-index error rate is nearly identical on the training and testing data as expected.  Don't know what the problem is.  Both values are about 36% (concordance = 100-36=64%).

**Reply by LukasBocek98** [reply]
Sorry if I was misspecified something, the OOB performance is indeed very well predicting the test performance. But I was expecting the in-bag error to be very similar to the OOB error...maybe that was wrong assumption from my side?

**Comment by ishwaran** [comment, marked as answer]
No, you should never use the inbag error for assessing prediction error performance.

Please see the Introduction to the following https://www.randomforestsrc.org/articles/forestWgt.html

**Reply by LukasBocek98** [reply]
Understand, thank you!
